### PR TITLE
[FLINK-36408] Fix MySQL pipeline connector failed to parse FLOAT type with precision

### PR DESCRIPTION
**Pull request by yuxiqian**
Currently, MySQL SourceRecordEmitter treats `FLOAT(x)` (float with explicit precision) as a `DOUBLE`, but pipeline `MetadataAccessor` reports a plain "FLOAT", which causes discrepancy and corrupted deserialized data.

This bug should be fixed by adjusting `MetadataAccessor` logic (mapping `FLOAT(x)` to `DOUBLE`).

### CODE REVIEW

**Comment by yuxiqian** [comment]
cc @ruanhang1993

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/utils/MySqlTypeUtils.java [hunk 1]
```diff
@@ -164,7 +164,12 @@ private static DataType convertFromColumn(Column column) {
             case FLOAT:
             case FLOAT_UNSIGNED:
             case FLOAT_UNSIGNED_ZEROFILL:
-                return DataTypes.FLOAT();
+                if (column.length() != -1) {
```
Comment (ruanhang1993) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/utils/MySqlTypeUtils.java:167]: Please add a constant for the default value -1.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/utils/MySqlTypeUtils.java [hunk 2]
```diff
@@ -164,7 +165,12 @@ private static DataType convertFromColumn(Column column) {
             case FLOAT:
             case FLOAT_UNSIGNED:
             case FLOAT_UNSIGNED_ZEROFILL:
-                return DataTypes.FLOAT();
+                if (column.length() != FLOAT_LENGTH_UNSPECIFIED_FLAG) {
+                    // For FLOAT types with length provided explicitly, treat it like DOUBLE
+                    return DataTypes.DOUBLE();
+                } else {
+                    return DataTypes.FLOAT();
+                }
```
Comment (ruanhang1993) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/utils/MySqlTypeUtils.java:173]: We should change the documents about the data type mapping.

**Review by ruanhang1993 — APPROVED** [review]
LGTM

**Comment by yuxiqian** [comment]
Rebased #3696 into this, could @ruanhang1993 please trigger CI again?

**Review by ruanhang1993 — APPROVED** [review]
LGTM.

**Pull request merged**